src/app/icd/ICDCheckInSender.cpp

```diff
@@ -61,7 +58,14 @@ CHIP_ERROR ICDCheckInSender::SendCheckInMsg(const Transport::PeerAddress & addr)
     VerifyOrReturnError(!buffer.IsNull(), CHIP_ERROR_NO_MEMORY);
     MutableByteSpan output{ buffer->Start(), buffer->MaxDataLength() };
 
-    ReturnErrorOnFailure(CheckinMessage::GenerateCheckinMessagePayload(mAesKeyHandle, mICDCounter, ByteSpan(), output));
+    // Encoded ActiveModeThreshold in littleEndian for Check-In message application data
+    uint8_t activeModeThresholdBuffer[2] = {};
+    ByteSpan activeModeThresholdByteSpan(activeModeThresholdBuffer);
+    Encoding::LittleEndian::Put16((uint8_t *) activeModeThresholdByteSpan.data(),
+                                  ICDConfigurationData::GetInstance().GetActiveModeThresholdMs());
```

Comment:
```suggestion
    uint8_t activeModeThresholdBuffer[2] = {};
    Encoding::LittleEndian::Put16(activeModeThresholdBuffer,
                                  ICDConfigurationData::GetInstance().GetActiveModeThresholdMs());
    ByteSpan activeModeThresholdByteSpan(activeModeThresholdBuffer);
```

Reply:
And in particular, hiding a const_cast inside a C-style cast is just bad.  We should never do that. If we need a const_cast we should have a const_cast.  But in this case we don't need it.